app/app_test.go

```diff
@@ -129,6 +129,35 @@ func TestThriftDecoding(t *testing.T) {
 	assert.Equal(w.Code, http.StatusAccepted)
 }
 
+func TestThriftRootSpans(t *testing.T) {
+	// Test that spans with a *zero* parentID get converted to spans with a nil
+	// parentID.
+	assert := assert.New(t)
+	var zero int64
+	body := serializeThriftSpans([]*zipkincore.Span{
+		&zipkincore.Span{
+			TraceID:  2222,
+			ID:       2222,
+			ParentID: &zero,
```

Comment:
I don't really understand the pointer type here - would this test fail if `ParentID: 0` instead of a `*int64`?

Reply:
In the thrift serialization format, the `ParentID` field can be zero, non-zero, or nil. So the source struct field has to have a pointer type to account for all three possibilities (it's defined over here https://github.com/jaegertracing/jaeger-client-go/blob/master/thrift-gen/zipkincore/ttypes.go#L715)